homeassistant/components/light/hue.py

```diff
@@ -174,25 +173,28 @@ def update_lights():
         else:
             bridge_type = 'hue'
 
-        for light_id, info in api_lights.items():
-            if light_id not in lights:
-                lights[light_id] = HueLight(int(light_id), info,
-                                            bridge, update_lights,
-                                            bridge_type, allow_unreachable)
-                new_lights.append(lights[light_id])
-            else:
-                lights[light_id].info = info
-                lights[light_id].schedule_update_ha_state()
-
-        for lightgroup_id, info in api_groups.items():
-            if lightgroup_id not in lightgroups:
-                lightgroups[lightgroup_id] = HueLight(
-                    int(lightgroup_id), info, bridge, update_lights,
-                    bridge_type, allow_unreachable, True)
-                new_lights.append(lightgroups[lightgroup_id])
-            else:
-                lightgroups[lightgroup_id].info = info
-                lightgroups[lightgroup_id].schedule_update_ha_state()
+        for is_group, fetched, current in (
+                (False, api_lights.items(), lights),
+                (True, api_groups.items(), lightgroups)):
+
+            for light_id, info in fetched:
+                light = current.get(light_id)
+
+                is_new = False
+                if light is None:
+                    is_new = True
+                    light = current[light_id] = HueLight(
+                        int(light_id), bridge, update_lights,
+                        bridge_type, allow_unreachable, is_group)
+                    new_lights.append(light)
+
+                if is_group:
+                    light.info = info['action']
```

Comment:
This will break `def unique_id` and `def name`, since it needs `self.info.get('uniqueid', self.name))`. You still need the real `info` dict for those functions.